server/src/internalClusterTest/java/org/opensearch/nodestats/NodeStatsIT.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.nodestats;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.DocWriteResponse;
+import org.opensearch.action.bulk.BulkItemResponse;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.index.engine.DocumentMissingException;
+import org.opensearch.index.engine.VersionConflictEngineException;
+import org.opensearch.index.shard.IndexingStats.Stats.DocStatusStats;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchIntegTestCase.ClusterScope;
+import org.opensearch.test.OpenSearchIntegTestCase.Scope;
+import org.hamcrest.MatcherAssert;
+
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+@ClusterScope(scope = Scope.SUITE, numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)
+public class NodeStatsIT extends OpenSearchIntegTestCase {
+
+    private static final DocStatusStats expectedDocStatusStats = new DocStatusStats();
```

Comment:
Can you ensure this object gets reset in @Before for every test.

Reply:
It's an integration test. Cluster's state is maintained throughout the test run. Using `expectedDocStatusStats`, we're manually accounting all the doc statuses while `assertDocStatusStats()` is used to fetch the actual stats counted by the cluster.

Can we do it at individual test-level basis?

Reply:
Each test should be contained in itself and start from clean state. If a previous test failed half way, it shouldnt impact other tests.

Reply:
- Made `expectedDocStatusStats` non-static
- reduced scope to `Scope.TEST` from `Scope.SUITE`
```diff
-@ClusterScope(scope = Scope.SUITE, ...)
+@ClusterScope(scope = Scope.TEST,...)
```